lib/diagnostic.rb

```diff
@@ -7,37 +7,55 @@ class Diagnostic
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    the_length
+    the_longest_string
+    the_longest_string = list.max_by(&:length)
+    the_length = the_longest_string.length # find length of longest string returned
+    return the_length
   end
+#Source: Slackoverflow. Tested in pry
 
   # Question 2
   # Write ruby code to return the sum of entries in the
   # list supplied (i.e. apply the `+` operator).
   # Assume each entry implements a `+` method.
   def sum(list)
-    # Your code goes here
+    list.inject(0){|sum,x| sum + x }
   end
 
+  #Source: Slackoverflow. Tested in pry
+
   # Question 3
   # Write ruby code that returns true if some of the entries
   # in the list of numbers are odd
+  # I assumed list was size 10.
   def some_odd(numbers)
-    # Your code goes here
+    x = 1
+    returnValue = false
+    loop do
+      x += 1
+      next unless ( x % 2 ) != 0 # find the odd number
+      returnValue = true
+      break if x >= 10
+    end
+    return returnValue
```

Comment:
Remember here we don't need the keyword `return`. Here's the quick Ruby way to do this problem:

`numbers.any?(&:odd?)`